### PR TITLE
Ensure that all of the DataType constants are actually of type DataType

### PR DESCRIPTION
**Pull request by jsternberg**

### CODE REVIEW

File: ast.go [hunk 1]
```diff
@@ -23,23 +23,23 @@ const (
...
+	Duration DataType = 6
 	// Tag means the data type is a tag.
-	Tag = 7
+	Tag DataType = 7
 	// AnyField means the data type is any field.
-	AnyField = 8
+	AnyField DataType = 8
 	// Unsigned means the data type is an unsigned integer.
-	Unsigned = 9
+	Unsigned DataType = 9
```
Comment (stuartcarnie) [inline ast.go]: is there any reason we don't use `iota` and remove the explicit assignment of numbers?

Comment (jsternberg) [reply ast.go]: It ensures we always use the same numbers and don't accidentally change them.

Comment (stuartcarnie) [reply ast.go]: Do we serialize the values? 

We can ensure that same behavior with `iota` by always adding new constants to the end and verifying in code review.

Comment (jsternberg) [reply ast.go]: We serialize the values. While it's possible to use `iota`, I would be more comfortable if we just specified the values explicitly.

**Pull request merged**